src/compute-client/src/controller/instance.rs

```diff
@@ -341,22 +340,7 @@ impl<T: ComputeControllerTimestamp> Instance<T> {
     }
 
     fn remove_replica_state(&mut self, id: ReplicaId) -> Option<ReplicaState<T>> {
-        let Some(replica) = self.replicas.remove(&id) else {
-            return None;
-        };
-
-        if let Some(time) = replica.last_heartbeat {
-            let row = Row::pack_slice(&[
-                Datum::String(&id.to_string()),
-                Datum::TimestampTz(time.try_into().expect("must fit")),
-            ]);
-            self.deliver_introspection_updates(
-                IntrospectionType::ComputeReplicaHeartbeats,
-                vec![(row, -1)],
-            );
-        }
-
-        Some(replica)
+        self.replicas.remove(&id)
     }
```

Comment:
Nit: Now that this method only consists of a single line, how about inlining it in the only place where it is used?

Reply:
Works for me!